widgetssdk/src/main/java/com/glia/widgets/core/secureconversations/domain/GetUnreadMessagesCountWithTimeoutUseCase.kt

```diff
@@ -30,7 +36,13 @@ internal class GetUnreadMessagesCountWithTimeoutUseCase(private val repository:
      * there is no warranty that it will return anything so timeout is added to make sure that it will return [NO_UNREAD_MESSAGES]
      * after the timeout if there is no answer from socket yet.
      */
-    operator fun invoke(): Single<Int> = Single.create {
+    operator fun invoke(): Single<Int> = when {
+        core.isInitialized.not() -> Single.just(NO_UNREAD_MESSAGES)
+        isAuthenticatedUseCase().not() -> Single.just(NO_UNREAD_MESSAGES)
+        else -> getUnreadMessagesCountWithTimeout()
```

Comment:
Although it can also work well, I think that it's not perfect to return `NO_UNREAD_MESSAGES` for Glia not initialized and visitor not authenticated cases because, in fact, we don't know this rather than there are no unread messages. Ideally, `GetUnreadMessagesCountWithTimeoutUseCase` should return an error. This would allow the caller to distinguish when to show "messaging unavailable" vs "no unread messages".

But I'm not strongly against the solution that you implemented. I just want to bring my concern up to know if you took this into account.

Reply:
What will give it to us?
We will have to map the error, regardless of the type, to the default value in all the places we use these use cases.
I mean, these functions are designed for the use cases where we always need a value regardless of the request result. So, if we want to deal with errors, we can still use the repository functions.

Reply:
Okay, maybe this is worth adding as a comment: 
> if we want to deal with errors, we can still use the repository functions.

and add references to the functions

Reply:
added